tests/integration/tier0/test-check-rollback-handling/test_check_rollback_handling.py

```diff
@@ -104,3 +101,111 @@ def test_check_untrack_pkgs_force(convert2rhel, shell):
         assert c2r.exitstatus != 0
 
     post_rollback_check(shell)
+
+
+def terminate_and_assert_good_rollback(c2r):
+    if "oracle-7" in booted_os or "centos-7" in booted_os:
+        # Use 'Ctrl + c' first to check for unexpected behaviour
+        # of the rollback feature after process termination
+        c2r.sendcontrol("c")
+        # Due to inconsistent behaviour of 'Ctrl + c' on some distros
+        # use Ctrl + d instead
+        c2r.sendcontrol("d")
+        # Assert the rollback finished all tasks by going through its last task
+        assert c2r.expect("Rollback: Removing installed RHSM certificate") == 0
+        assert c2r.exitstatus != 1
+    else:
+        c2r.sendcontrol("c")
+        assert c2r.expect("Rollback: Removing installed RHSM certificate") == 0
+        assert c2r.exitstatus != 1
+
+
+def test_terminate_registration_asap(convert2rhel):
```

Comment:
I do not like the word asap used here. I would use something like:
`test_terminate_on_registration_start`

Reply:
Good point, done.